pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -100,53 +102,162 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
+	persistentVolumeClaim, err := newPostgresqlPersistentVolumeClaim(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
 	// Set UnifiedPushServer instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, persistentVolumeClaim, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+	// Check if this PersistentVolumeClaim already exists
+	foundPersistentVolumeClaim := &corev1.PersistentVolumeClaim{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: persistentVolumeClaim.Name, Namespace: persistentVolumeClaim.Namespace}, foundPersistentVolumeClaim)
 	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
+		reqLogger.Info("Creating a new PersistentVolumeClaim", "PersistentVolumeClaim.Namespace", persistentVolumeClaim.Namespace, "PersistentVolumeClaim.Name", persistentVolumeClaim.Name)
+		err = r.client.Create(context.TODO(), persistentVolumeClaim)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
 
-		// Pod created successfully - don't requeue
-		return reconcile.Result{}, nil
+	postgresqlDeployment, err := newPostgresqlDeployment(instance)
```

Comment:
From here to L150 is basically a copypasta of the above logic, and from L150 down it's mostly just repeating the same-ish logic over and over for each resource that's being created. I spent a while trying to create a generic function that could do it for all objects, but I was unsuccessful (my Go-fu is not strong enough).